### ISSUE TITLE
List view of projects

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/action#966]
## Issue - Enhancement

Background: Good OOUX separate the concern of system objects from how and when they are presented (which is JBTD, etc.). Having a list view of a subset of projects, and a list-item presentation of individual projects, may facilitate different workflows, like processing inbox (unprocessed) content.

**Estimated effort**: algunos puntos ([see CONTRIBUTING.md](https://github.com/ParabolInc/action/blob/master/CONTRIBUTING.md#points-and-sizes))

**Comment by ackernaut** [comment]
See #980

**Comment by ackernaut** [comment]
Going to expand on some recent ideas here:

List view is a way to process a batch of cards that could serve many UX cases aka JTBDs:

- A batch of cards comes in because of a new integration (fitting the Inbox UX)
- A batch of cards had an integration removed (fitting either the Import/Switching UX or a teammate and sole integrator left the team, or just the sole integration was removed)
- A teammate was removed, or left, and the team lead needs to re-prioritize and re-assign the inherited projects/tasks
- Perhaps in the future, a batch of cards could be reviewed in list style for employee review, team review, or retro?

The list view is a larger module, just like we could consider the single kanban (the new dashboard) a larger module.

Some initial thoughts and details (some future, others sooner (Epic 8?):

- The list view can be triggered with various options to fit the UX. The options can involve what filtering, sorting and actions are available in that context.
- Sorting (options): created at (newest/oldest), updated at (most/least recent), any more?
- Actions (options): snooze, archive, make private, raise as agenda, set status [I/S/A/D], assign, add tag, export, make report
- Filters (options): status, owner, team, tag, event, source/provider?
- Dashboard has a toggle to get back to kanban view
- Card backs can open up over LIST VIEW (when users want to know more about a listed project/task and we have deep card backs), but the permalink for a card opens up over kanban view?
- Simple list views for specific cases may even fit in a panel, a modal, etc. but the utility would probably feel more familiar as a function of dashboard.

List view can be triggered by different work flows:

- Tap LIST VIEW on a kanban column switches to list view filtered by that status (esp. INBOX)
- Tap event (via notification, email?) shows LIST VIEW for the cards affected by the event

**Events involving cards**
Here’s a first pass at a spec:
- Has a message “something happened regarding a teammate, an integration, etc.”
- Has an action “Do Something About These Cards” (that opens to list view including the cards associated with the event)
- Perhaps events can be active/current OR cleared/resolved
- Perhaps users can always see a history of past events in under a cleared/resolved tab
- Maybe the primary channel for events is in user notifications? or follows the notification system however we design it if we do indeed introduce settings/throttles for notifications, emails, subscriptions, etc.
- I think users can always tap the event action to see the list and the cards will show in their most current state (so if status, owner, archiving has changed, well—they see it!)

**Comment by ackernaut** [comment]
@jordanh 👀  thoughts?

**Comment by jordanh** [comment]
This is a good start!

As @mattkrick and I mentioned, I can't see a JTBD involving removing an integration (is there something we are missing?)

I think the primary JTBD for list view will be applying bulk operations to a bunch of tasks. Select multiple should be the primary operation, followed by "apply operation to X items"

**Comment by ackernaut** [comment]
Ha! Yeah, I guess the JTBD happens “when an integration is removed” “I want [something]” “so that I can resolve what to do with these cards”. The JTBD is not “when [situation]” “I want to remove an integration” “so [desired outcome]”.

I guess it boils down to how and when an integration might be removed…even though they do not represent primary UX for our value add. If I can connect GitHub, I can disconnect GitHub.

Say X number of folks have cards assigned to them. All of a sudden they are gone, archived, because 1 person removed the integration? “What happened to my cards?”

That does not give me the impression of team clarity.

Basically, I think I am overcompensating for folks knowing what happens when it happens.

Perhaps my minimal ask here would be: If a GH integration is removed, and cards are archived, each person who had cards (or maybe the whole team) gets a notification: “The GitHub repository **repo-name** was removed. **N** cards were archived.”

In the future it might make sense to see that list of cards on the single kanban (by #archived by #eventId (the integration removal) so I could know what to do about it.

I could be wrong on this one and not saying anything is unsafe to try. :)

**Comment by jordanh** [comment]
I think there is good chance that you might be correct. Could we perhaps make this a bit safer by sending a notification to the team members that says, "GitHub integration removed, X cards archived," as a starting point to cover the JTBD?

**Comment by ackernaut** [comment]
Yeah, I like the simple notification idea. We’ve mentioned it here and in #1155.

**Comment by ackernaut** [comment]
I like how GitHub presents Sorting like a type of filter:

![screenshot 2017-07-27 15 28 50](https://user-images.githubusercontent.com/307286/28690882-66c9ed78-72e0-11e7-9e16-2f50234c8474.png)
